### PR TITLE
Downgrade compiler to CentOS 4.8.5

### PR DESCRIPTION
**Pull request by yuslepukhin**
**Description**:  Support customers on CentOS

**Motivation and Context**
Make CPU build compile and run on CentOS 7 with GCC 4.8.5

Initially I thought of setting -std=c++11, however, this turned out to be a bad idea.
CentOS compiler ignores the setting but all others downgrade and start complaining about certain C++14 features that CentOS actually supports. Case in point local var lambda captures which I thought would be unfortunate to loose.

Long story short, there is not a specific standard the CentOS GCC supports.

### CODE REVIEW

**Comment by snnn** [comment]
In file included from /onnxruntime_src/onnxruntime/test/automl_ops/datetimetransformer_test.cc:5:0:
/onnxruntime_src/onnxruntime/test/providers/provider_test_utils.h: In instantiation of 'void onnxruntime::test::OpTester::AddOutput(const char*, T&&) [with T = Microsoft::Featurizer::DateTimeFeaturizer::TimePoint]':
/onnxruntime_src/onnxruntime/test/automl_ops/datetimetransformer_test.cc:38:52:   required from here
/onnxruntime_src/onnxruntime/test/providers/provider_test_utils.h:261:116: error: no matching function for call to 'onnxruntime::test::OpTester::Data::Data(onnxruntime::NodeArg, OrtValue&, onnxruntime::test::optional<float>, onnxruntime::test::optional<float>)'
     output_data_.push_back(Data(NodeArg(name, mltype->GetTypeProto()), value, optional<float>(), optional<float>()));
                                                                                                                    ^
/onnxruntime_src/onnxruntime/test/providers/provider_test_utils.h:261:116: note: candidate is:
/onnxruntime_src/onnxruntime/test/providers/provider_test_utils.h:308:10: note: onnxruntime::test::OpTester::Data::Data(onnxruntime::test::OpTester::Data&&)
   struct Data {
          ^
/onnxruntime_src/onnxruntime/test/providers/provider_test_utils.h:308:10: note:   candidate expects 1 argument, 4 provided

File: cmake/CMakeLists.txt [hunk 1]
```diff
@@ -12,8 +12,11 @@ set(CMAKE_POLICY_DEFAULT_CMP0069 NEW)
 include(CheckCXXCompilerFlag)
 include(CheckLanguage)
 
-# Set C++11 as standard for the whole project
-set(CMAKE_CXX_STANDARD 11)
+# CentOS compiler is old but it does allow certain C++14 features
+# such as lambda captures and they are convinient
```
Comment (snnn) [inline cmake/CMakeLists.txt]: # g++ --std=c++14
g++: error: unrecognized command line option '--std=c++14'

Comment (yuslepukhin) [reply cmake/CMakeLists.txt]: This is actually the original code, not my invention. The option has one leading dash.
